python/cudf/cudf/_lib/pylibcudf/copying.pxd

```diff
@@ -9,20 +9,36 @@ from .column cimport Column
 from .scalar cimport Scalar
 from .table cimport Table
 
+ctypedef fused ColumnOrTable:
+    Table
+    Column
+
+
+ctypedef fused TableOrListOfScalars:
+    Table
+    # The contents of the list must be validated as Scalars at runtime.
+    list
+
+
+# Need two separate fused types to generate the cartesian product of signatures.
+ctypedef fused LeftCopyIfElseOperand:
+    Column
+    Scalar
+
+ctypedef fused RightCopyIfElseOperand:
+    Column
+    Scalar
```

Comment:
For other reviewers: this took me a bit by surprise but it's the documented way to generate the cross product of types (see around [this](https://cython.readthedocs.io/en/latest/src/userguide/fusedtypes.html#fused-types-and-arrays) part of the docs).

Reply:
Should we generalize the name here a bit more: `LeftColumnOrScalar` and `RightColumnOrScalar`?

Reply:
It's tricky. In theory these typedefs are identical to the ones used for binary ops, but my first attempt at sharing them using a more generic name and sharing them failed because I put them in types.pxd and that caused circular import issues (importing column/scalar into types). Do you think it makes sense to create a separate `fused_types.pxd` header for this purpose?

Reply:
Ah, maybe we punt until the need arises then.

Reply:
OK sounds good. If we start seeing the same fused types appearing many times I'll make a shared Cython header just for those.